.github/workflows/apply-template.yml

```diff
@@ -0,0 +1,68 @@
+name: Apply Template
+
+on:
+  push:
+    branches: ["main"]
+
+jobs:
+  repo_ids:
+    name: Get Repository Name
+    runs-on: ubuntu-latest
+    outputs:
+      is_renamed: ${{ steps.check_rename_occurred.outputs.IS_RENAMED }}
+      repo_name: ${{ steps.check_rename_occurred.outputs.REPO_NAME }}
+      org_name: ${{ steps.check_rename_occurred.outputs.ORG_NAME }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check if rename has occurred
+        id: check_rename_occurred
+        run: |
+          LINES=$(grep openapi-ts-template ./package.json)
+          if [ -z "$LINES" ]; then
+            echo "IS_RENAMED" >> true"
+          else
+            echo "IS_RENAMED" >> false"
+          fi
+          REPO_NAME=$(echo "${{ github.event.repository.name }}" | tr '[:upper:]' '[:lower:]')
+          ORG_NAME=$(echo "${{ github.event.repository.owner.name }}" | tr '[:upper:]' '[:lower:]')
+          echo "REPO_NAME >> $REPO_NAME"
+          echo "ORG_NAME >> $ORG_NAME"
+  update-repo:
+    name: Update repository references
+    runs-on: ubuntu-latest
+    needs: [repo_ids]
+    if: ${{ github.event.repository.name != 'openapi-ts-template' && needs.repo_ids.outputs.is_renamed == 'false' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install yq
+        run: sudo snap install yq
+      - name: Reset versioning
+        run: |
+          npm version 0.0.1 --allow-same-version --no-git-tag-version
+      - name: Update repo name references
+        run: |
+          find . \( -path ./.git -prune -o -path ./.github/workflows -prune \) -type f -o -name '*' -exec sed -i 's/openapi-ts-template/${{ needs.repo_ids.outputs.repo_name }}/gI' {} \;
+          find . \( -path ./.git -prune -o -path ./.github/workflows -prune \) -type f -o -name '*' -exec sed -i 's/digicatapult/${{ needs.repo_ids.outputs.org_name }}/gI' {} \;
+      - name: Remove template workflow
+        run: rm ./.github/workflows/apply-template.yml
+      - name: Set Git Config
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+      - name: Commit changes
+        run: |
+          git add -A
+          git commit -m "Apply templating to ${{ needs.repo_ids.outputs.repo_name }}"
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
```

Comment:
there's a `v4` for this action now which uses `node 16` rather than deprecated `node 12`